designs/2021-init-command-eslint-cli/README.md

```diff
@@ -0,0 +1,140 @@
+- Repo: [eslint/eslint](https://github.com/eslint/eslint)
+- Start Date: 2020-06-29
+- RFC PR:
+- Authors: Aniketh Saha ([anikethsaha](https://github.com/anikethsaha)) aladdin-add(weiran.zsd@outlook.com)
+
+# Move --init flag into a separate utility
+
+## Summary
+
+- Move the `init` command from main repo ([eslint](https://github.com/eslint/eslint)) to a new repo named `@eslint/create`.
+- Remove the auto-config.
+
+## Motivation
+
+Currently the whole `init` command is being shipped with the cli in the main repo. Though its not that much of size (roughly ~0.6MB for the [eslint/lib/init](https://github.com/eslint/eslint/tree/master/lib/init)), this command is not a type of command that we need everyday or everytime running eslint. It is mainly used when creating a new project or adding eslint to a project for the first time. So if we move this to a separate package that is meant to use in the command line,
+We will make use of tools like `npx` to simply run `npx @eslint/create` or using `npm` to run `npm init @eslint` or using `yarn` to run `yarn create @eslint` single time for a project instead of having it in the core project.
+
+## Detailed Design
+
+<!--
+   This is the bulk of the RFC.
+
+   Explain the design with enough detail that someone familiar with ESLint
+   can implement it by reading this document. Please get into specifics
+   of your approach, corner cases, and examples of how the change will be
+   used. Be sure to define any new terms in this section.
+-->
+
+The implementation is mainly migrating the code from main repo to `@eslint/create`. As far as the modules are concerned, all shared modules are being shipped currently with the eslint so we can use those modules through eslint itself.
+
+The approach would be following these steps
+
+### Move `eslint --init` related files to a separate repo
+
+- Move the `eslint/lib/init/*` to the new repo's `lib/*` directory
+- Add the following `dependencies` in `package.json`(the last 3 dependencies will be removed in next step)
+
+  - `enquirer`
+  - `progress`
+  - `semver`
+  - `lodash`
+  - `json-stable-stringify-without-jsonify`
+  - `cross-spawn`
+  - `debug`
+  - `@eslint/eslintrc` (to be removed)
+  - `eslint` (to be removed)
+  - `espree` (to be removed)
+
+- for all those modules coming inside `../**/*`, it would be replaced using `eslint/**/*`
+- For `tests`, move the `tests/lib/init` to new repo's `tests/init`
+- Add the following `devDependencies` in `package.json`
+
+  - `chai`
+  - `sinon`
+  - `js-yaml`
+  - `proxyquire`
+  - `shelljs`
+
+It was almost done: [aladdin-add/eslint-create](https://github.com/aladdin-add/eslint-create).
```

Comment:
It’s better to start a new repo from scratch for copyright reasons.

Reply:
for the moment, I have to do so.  As I don't have the access to create a new repo in eslint group.

Reply:
I can do it when the time comes. My point is that there are copyright issues with transferring an external repo into the eslint org. It would be better to wait until we setup a repo with appropriate license and copyright notices and then send a pull request there.